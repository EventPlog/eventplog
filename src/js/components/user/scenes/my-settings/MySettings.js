import React, { Component } from 'react'
import { Grid, Menu, Segment } from 'semantic-ui-react'
import styled from 'styled-components'

// =========== INTERNAL ============
import EditUser from 'js/components/user/scenes/edit-user'
import { mixins } from 'js/styles/mixins'
import sharedStyles from '../sharedStyles'

const StyledSettings = styled.div`
  ${sharedStyles}
`

import Auth from 'js/auth'

const MyResources = ({
  user,
  currentUser,
  resources,
  labelVerbMapping = [],
  activeItem,
  handleItemClick,
}) => {
  return (
    <StyledSettings>
      <Grid className='app-container'>
        <Grid.Column width={4}>
          <Menu fluid vertical tabular>
            {
              Object.keys(labelVerbMapping).map(item => (
                <Menu.Item
                  name={item}
                  active={activeItem === item}
                  onClick={handleItemClick}
                />
              ))
            }
          </Menu>
        </Grid.Column>

        <Grid.Column stretched width={12}>
          <Segment>
            <EditUser/>
          </Segment>
        </Grid.Column>
      </Grid>
    </StyledSettings>
  )
}

export default MyResources