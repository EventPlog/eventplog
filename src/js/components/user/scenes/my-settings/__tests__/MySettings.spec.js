import React from 'react';
import MySettings from '../MySettings';
import EditUser from 'js/components/user/scenes/edit-user'
import { shallow } from 'enzyme';

describe('MySettings', () => {
  it('should render correctly', () => {
    const wrapper = shallow( <MySettings  /> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(EditUser).length).toEqual(1);
  })
});