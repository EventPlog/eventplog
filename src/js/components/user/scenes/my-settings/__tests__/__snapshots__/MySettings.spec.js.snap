// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySettings should render correctly 1`] = `
<styled.div>
  <Grid
    className="app-container"
  >
    <GridColumn
      width={4}
    >
      <Menu
        fluid={true}
        tabular={true}
        vertical={true}
      />
    </GridColumn>
    <GridColumn
      stretched={true}
      width={12}
    >
      <Segment>
        <Component />
      </Segment>
    </GridColumn>
  </Grid>
</styled.div>
`;
